contracts/Vault.vy

```diff
@@ -870,9 +870,8 @@ def _shareValue(shares: uint256) -> uint256:
         # NOTE: if sqrt(Vault.totalAssets()) >>> 1e39, this could potentially revert
     lockedFundsRatio: uint256 = (block.timestamp - self.lastReport) * self.lockedProfitDegration
     freeFunds: uint256 = self._totalAssets() 
-    if(lockedFundsRatio < convert(1e18, uint256)):
-        freeFunds = freeFunds - (self.lockedProfit - (lockedFundsRatio * self.lockedProfit / convert(1e18, uint256)))
-    
+    locked:uint256 = self.lockedProfit - (lockedFundsRatio * self.lockedProfit / convert(1e18, uint256))
+    freeFunds = min(freeFunds, freeFunds - locked)
```

Comment:
```suggestion
    freeFunds -= locked
```

Reply:
makes sense, so no need for comparison. We can always safely substract the locked funds since they will be eventually zero within the time frame

Reply:
Can't the value turn negative and overflow?

Reply:
if you revert back to min does it make the pass? i think you are right

Reply:
I added it back already